pkg/openwsn/contrib/eui64.c

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2017 Hamburg University of Applied Sciences
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     pkg_openwsn
+ * @{
+ *
+ * @file
+ * @brief       RIOT adaption definition of the "eui64" bsp module
+ *
+ * @author      Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include "string.h"
+#include "eui64.h"
+
+#include "luid.h"
+#include "net/ieee802154.h"
+
+void eui64_get(uint8_t *addressToWrite)
+{
+    luid_get((void *)addressToWrite, IEEE802154_LONG_ADDRESS_LEN);
```

Comment:
This will get you a different EUI-64 everytime the function is called or is this intended (couldn't find any documentation, on this but I saw it is called [twice here](https://github.com/openwsn-berkeley/openwsn-fw/blob/2aa270d1da7fb2bb573372d994d1e3cca041288b/openapps/cjoin/cjoin.c#L88-L90))?

If not, why not get the long-address from the device somehow? E.g. have

```C
netdev_t *openwsn_netdev;

void eui64_get(uint8_t *addressToWrite)
{
    eui64_t eui64;
    if (openwsn_netdev->driver->get(dev, NETOPT_ADDRESS_LONG, &eui64, sizeof(eui64_t)) == sizeof(eui64_t)) {
        memcpy(addressToWrite, eui64.u8, sizeof(eui64.u8));
    }
    else {
        /* Use your code as a fallback or print an error message here */
        luid_get((void *)addressToWrite, IEEE802154_LONG_ADDRESS_LEN);
    }
}
```

And set netdev in `radio_init()`:

```C
extern netdev_t *openwsn_netdev;

void radio_init(void)
{
    /* ... */
#ifdef MODULE_AT86RF2XX
    openwsn_netdev = (netdev_t *)&at86rf2xx_dev.netdev.netdev;
    radio_vars.dev = openwsn_netdev;
    at86rf2xx_setup(&at86rf2xx_dev, &at86rf2xx_params[0]);
#endif
```

If you fear memory waste, take `netdev` out of `radio_vars_t` and use `openwsn_netdev` instead of `radio_vars.dev`.

/cc @fjmolinas